flight / telemetry serial configuration work

mlyle opened the pull request:
Fixes #600 
- Correct pipxtreme buffer size and case statement
- Make a common parent type for almost all baud rate information.
- Increase default telemetry rate to 115200 in both flight and telemetry
- Factor baud rate management into PIOS_HAL from UAVOMSPBridge, UAVOMavlinkBridge, UAVOLightTelemetryBridge, Telemetry, PicoC, GPS, and ComUsbBridge.
- Teach baud rate management code how to initialize COTS bluetooth modules
- Adjust pipxtreme to always project a virtual "modulesettings"
- Eliminate raw-passthrough mode that is unused in radiocombridge

Also removes some i2c vm items that were missed and unused.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/d-ronin/dronin/607)

<!-- Reviewable:end -->

mlyle committed:
flight: pipxtreme: reindent

mlyle committed:
flight: pipxtreme: fix tlm bt buffer sizes

mlyle committed:
serial telemetry: bump default baud rate to 115200

Stub out beginning of support for initializing bluetooth modules and set
that as the default.  Change name of setting in GCS so that we pick up
new default value in upgrade.  Change python API to be aware of 115200
default.

mlyle committed:
modulesettings: remove i2c vm related items.

mlyle committed:
flight: telemetry: normalize indent in prep for further work

mlyle committed:
flight: baud rate management: major changes

* Make a common parent type for almost all baud rate information.
* Default telemetry sessions to 115200, because fast is good.
* Factor baud rate management into PIOS_HAL from UAVOMSPBridge,
UAVOMavlinkBridge, UAVOLightTelemetryBridge, Telemetry, PicoC, GPS, and
ComUsbBridge.
* Teach baud rate management code the beginnings of how to configure
bluetooth modules.
* Alter pipxtreme target to always communicate rates using a projected
ModuleSettings.
* Fixup the simulator to get a null pios_hal.

mlyle committed:
flight: radiocombridge: eliminate unused raw-passthrough mode

mlyle committed:
pios_hal: attempt to find init seq for bluetooth modules

mlyle commented:
This is WIP.  Untested and there's some more work to do, I think.

pug398 commented:
@mlyle Out of curiosity have you ever had 115k telemetry work?
I always thought there was something wrong (other than setting gcs serial rate to 115k).

mlyle commented:
@pug398 There have definitely been bugs in pipxtreme, I'm studying it now.

mlyle committed:
flight: serial: defer initialization

mlyle committed:
pios_hal: teach about HC-05 init sequence

mlyle commented:
- Normal telemetry works at 115200
- Telemetry works between Quanton and GCS at 115200 with HC-05; doing the init sequence requires some magic (halt board, hold down button on HC-05, power up).  Shows up properly, pin 0000, 115200 on subsequent boots.

Supposedly some HC-05s come up in AT mode so this would be a lot easier.  

```
Michaels-MBP:dronin mlyle$ ls -la /dev/tty.dRonin-DevB 
crw-rw-rw-  1 root  wheel   18,  66 Feb  4 01:43 /dev/tty.dRonin-DevB
```

mlyle commented:
jenkins, build artifacts please

mlyle committed:
radiocombridge: defer registering telemetry tx wdg flag

mlyle commented:
jenkins, build artifacts please

pug398 commented:
jenkins, build artifacts please

mlyle committed:
flight: pios_hal: get HC-06 init sequence working

mlyle commented:
With this, plugging in a HC-06 to quanton, setting UART1 to telemetry, and the baud rate to 'INIT HC06' just magically works.  A device called 'dRonin' appears that one can connect to with GCS.

mlyle commented:
Plugging in a PL2303 USB adapter to Quanton UART2, with UART2 set to telemetry and the default baudrate enabled (115200 in both GCS and board) appears to work.

mlyle commented:
Tested mainport pipxtreme to PL2303 115200.

Tested mainport HC06 init sequence to HC06 dongle.

mlyle commented:
Tested PL-2303 @ 115200 to Sparky1 Flexiport.

mlyle commented:
Tried default serial telemetry (PL-2303 @ 115200) to Sparky2 mainport.

mlyle commented:
Tried default serial telemetry (PL2303 @ 115200) to Revolution mainport.

mlyle commented:
Tested bluetooth HC-06 and initialization sequence (automatic 115200) to revolution flexiport

mlyle committed:
python: trivial correction to python api

mlyle commented:
Tested python API over bluetooth serial to revo flexi...

```
Michaels-MBP:dronin mlyle$ ./python/get_config.py -s /dev/cu.dRonin-DevB
badver ab
Disconnected
Disconnected
Handshake ackd
Handshake ackd
Connected
Connected
Connected
UAVO_ActuatorSettings(name=UAVO_ActuatorSettings, time=1454909636.87, uavo_id=0x3557036A, MotorInputOutputCurveFit=(1.0, 1.0), TimerUpdateFreq=(50, 50, 50, 50, 50, 50), ChannelMax=(2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000), ChannelNeutral=(1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000), ChannelMin=(1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000), TimerPwmResolution=(1MHz(0), 1MHz(0), 1MHz(0), 1MHz(0), 1MHz(0), 1MHz(0)), ChannelType=(PWM(0), PWM(0), PWM(0), PWM(0), PWM(0), PWM(0), PWM(0), PWM(0), PWM(0), PWM(0)), MotorsSpinWhileArmed=FALSE(0))
UAVO_AttitudeSettings(name=UAVO_AttitudeSettings, time=1454909636.99, uavo_id=0x9DB8D0A, MagKp=0.0500000007451, MagKi=9.99999974738e-05, AccelKp=0.0500000007451, AccelKi=9.99999974738e-05, AccelTau=0.10000000149, VertPositionTau=2.0, YawBiasRate=9.99999997475e-07, BoardRotation=(0, 0, 0), ZeroDuringArming=TRUE(1), BiasCorrectGyro=TRUE(1), FilterChoice=CCC(0))
...
```

mlyle committed:
flight: serial: 115200bps is not 57600bps

mlyle commented:
jenkins, test this please

mlyle commented:
jenkins, test this please

mlyle commented:
jenkins build artifacts please

ukmook commented:
Tested init HC06 on sparky 2. Set Init HC06 telemetry on Sparky2 & Oplink, Used RFm bind wizard.
Paired Bluetooth to PC Used Password 0000. Working Ok

mlyle committed:
flight: telemetry: better buffer mgmt (interim chgs)

mlyle commented:
Everything is happier with larger buffers, too, it seems.

pug398 commented:
jenkins build artifacts please

mlyle committed:
flight: serial: adjust hm-10 init sequence

Am able to confirm this puts an HM-10 into the correct mode via
interrogation of it.  Unfortunately the HM-10 I bought has dead
RF so will not be able to fully verify

mlyle commented:
This is as good as it is going to get for awhile.  From the last commit message:

Am able to confirm this puts an HM-10 into the correct mode via
interrogation of it.  Unfortunately the HM-10 I bought has dead
RF so will not be able to fully verify.

tracernz commented:
This looks really good. Are you happy @pug398?

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 9 of 18 files at r1, 6 of 9 files at r2, 1 of 1 files at r5, 3 of 4 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[flight/targets/pipxtreme/fw/pios_board.c, line 182 [r8]](https://reviewable.io:443/reviews/d-ronin/dronin/607#-KA8gugiHcLc1IKYupZ0:-KA8gugiHcLc1IKYupZ1:1392701758) ([raw file](https://github.com/d-ronin/dronin/blob/1c228bc0d3798bb444b5e5c901d4c0c34ea5ecfc/flight/targets/pipxtreme/fw/pios_board.c#L182)):_
Does this result in a flash write every boot? I think pipx and probably taulink use the internal stm32 flash so maybe not ideal. But it looks like we already did it before so okay..

---

_[ground/gcs/src/plugins/serialconnection/serialpluginconfiguration.cpp, line 75 [r8]](https://reviewable.io:443/reviews/d-ronin/dronin/607#-KA8iHvXGCjx5XVTKKnb:-KA8iHvXGCjx5XVTKKnc:463315781) ([raw file](https://github.com/d-ronin/dronin/blob/1c228bc0d3798bb444b5e5c901d4c0c34ea5ecfc/ground/gcs/src/plugins/serialconnection/serialpluginconfiguration.cpp#L75)):_
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/607#-:-KA8iViRIi9wUXEtk39G:-1122423043)_

<!-- Sent from Reviewable.io -->

mlyle commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

_[flight/targets/pipxtreme/fw/pios_board.c, line 182 [r8]](https://reviewable.io:443/reviews/d-ronin/dronin/607#-KA8gugiHcLc1IKYupZ0:-KA8lEY9SB_dsZy46qpe:2069587900) ([raw file](https://github.com/d-ronin/dronin/blob/1c228bc0d3798bb444b5e5c901d4c0c34ea5ecfc/flight/targets/pipxtreme/fw/pios_board.c#L182)):_
No, it's just a set, which corresponds to "apply".  You have to explicitly ask to save it or do ObjectPersistence stuff from GCS to save.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/607)_

<!-- Sent from Reviewable.io -->

pug398 commented:
Looks good for what I have tested. Both HC-05 and HC-06 connected directly to flex port and also to main port on OPLM. This adds 115k functionality to telemetry where it wasn't there before and the ability to auto initialize bt module. (module should initially be at default 9600 baud) 

I am on the fence though about setting gcs telemetry defaults to 115k. Deep down I think I would rather have them at 57k and have to change them to make use of 115k.

mlyle commented:
@pug398 -- why do you think it might be better to speak 57600 by default?

I think we have some bugs currently with high rates and telemetry.. but changing the default during a dev cycle will be good to shake them out.  And once we're done 115200 ought to perform better, was my rationale.

mlyle commented:
Got a message from @pug398-- just summarizing:

A lot of the radio links, you can configure serial to 115200, but they won't actually attain the rate underneath and things will be flaky.

While we attempt to prioritize the serial link and retry things-- as noted in #641 it does not always go well.

I think we should merge this and use the increased default rate to shake out more of the issues with telemetry.

pug398 commented:
Sounds good to me. You just beat me out while I was typing an answer.
